src/DurableTask.AzureStorage/Partitioning/TablePartitionManager.cs

```diff
@@ -0,0 +1,804 @@
+                            }
+                            if (isReleasedLease)
+                            {
+                                this.settings.Logger.LeaseRemoved(
+                                    this.storageAccountName,
+                                    this.settings.TaskHubName,
+                                    this.settings.WorkerId,
+                                    partition.RowKey,
+                                    "",//Token empty cause there is no token in table partition manager.
+                                    "");//Leasetype empty cause there is no leaseType in table partition manager.
+                            }
+                            if (isRenewedLease)
+                            {
+                                this.settings.Logger.LeaseRenewalResult(
+                                    this.storageAccountName,
+                                    this.settings.TaskHubName,
+                                    this.settings.WorkerId,
+                                    partition.RowKey,
+                                    true,
+                                    "",
+                                    "",
+                                    $"Succeessfully renewed the lease of {partition.RowKey} during shutdown.");
+                            }
+                            
+                        }
+                        //Exception will be thrown due to concurrency.
+                        //Dequeue loop will catch it and re-read the table immediately to get the latest ETag
+                        catch (RequestFailedException ex) when (ex.Status == 412)
+                        {
+                            this.settings.Logger.PartitionManagerInfo(
+                            this.storageAccountName,
+                            this.settings.TaskHubName,
+                            this.settings.WorkerId,
+                            partition.RowKey,
+                            $"Failed to update {partition.RowKey} due to concurrency. Re-read the table to get the latest Etag.");
+                            throw;
+                        }
+                        // Eat any exceptions during lease stealing.
+                        catch (Exception exception)
+                        {
+                            this.settings.Logger.PartitionManagerError(
+                                this.storageAccountName,
+                                this.settings.TaskHubName,
+                                this.settings.WorkerId,
+                                partition.RowKey,
+                                $"Error in shutdown lease {partition.RowKey} : {exception}"
+                                );
+                        }
+                    }
+
+                }
+                bool isReleasedAllLease = (ownershipLeasesNum == 0);
+                return isReleasedAllLease;
```

Comment:
nit: consider this alternative variable names
```suggestion
                bool releasedAllLeases = (ownershipLeasesNum == 0);
                return releasedAllLeases;
```